src/core/lightning/index.types.d.ts

```diff
@@ -19,3 +19,15 @@ interface IPaymentRequest {
   memo?: string
   isReward?: boolean
 }
+
+type AddInvoiceSelfArgs = {
+  walletId: WalletId
+  memo?: string
```

Comment:
Memo + amount shouldn't be optional. Input validation should happen _before_ calling the use-case method.
Args should not accept undefined should just default to `memo: ""`, and `satoshis: 0` or something like that

Reply:
Perhaps the `satoshis: 0` is also its own use case (`AddNoAmountInvoice`). After all it is its own `gql` mutation:
```
  lnNoAmountInvoiceCreateOnBehalfOfRecipient(
    input: LnNoAmountInvoiceCreateOnBehalfOfRecipientInput!
  ): LnNoAmountInvoicePayload!
```

Reply:
I see that you are differentiating the use case methods with `NoAmount` so just the `Args` types are not in sync with the methods.

Reply:
Agree